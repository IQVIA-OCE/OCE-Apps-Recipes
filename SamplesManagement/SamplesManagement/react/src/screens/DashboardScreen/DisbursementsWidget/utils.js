import moment from 'moment';
const DATE_FORMAT = 'YYYY-MM';
import Color from 'color';

const base_colors = [
  Color('#5899DA'),
  Color('#E8743B'),
  Color('#19A979'),
  Color('#ED4A7B'),
  Color('#945ECF'),
  Color('#13A4B4'),
  Color('#525DF4'),
  Color('#BF399E'),
  Color('#6C8893'),
  Color('#EE6868'),
  Color('#2F6497'),
];

export const generateColors = total_colors => {
  const colors = [];
  const gradient_step = (base_colors.length - 1) / total_colors;

  let gradient = 0;

  colors.push(base_colors[0].hex());

  for (let i = 1; i < total_colors - 1; i++) {
    gradient += gradient_step;

    const from_index = Math.floor(gradient);
    let to_index = Math.ceil(gradient);

    if (from_index === to_index) {
      to_index = from_index + 1;
    }

    const from_color = base_colors[from_index];
    const to_color = base_colors[to_index];

    const mix_position = gradient % 1;

    colors.push(from_color.mix(to_color, mix_position).hex());
  }

  colors.push(base_colors[base_colors.length - 1].hex());

  return colors;
};

const generateDumpChartData = period => {
  const dumpData = [];
  for (let i = period; i >= 0; i -= 1) {
    dumpData.push({
      x: moment()
        .subtract(i, 'months')
        .format(DATE_FORMAT),
      y: 0,
    });
  }

  return dumpData;
};
export const normalizeDisbursements = period => data => {
  const normalized = data.reduce(
    (acc, disbursement) => {
      if (acc.allIds.indexOf(disbursement.Id) === -1) {
        acc.allIds.push(disbursement.Id);
      }

      if (!acc.byId[disbursement.Id]) {
        acc.byId[disbursement.Id] = {
          label: disbursement.productName ? disbursement.productName : '',
          data: generateDumpChartData(period),
        };
      }

      acc.byId[disbursement.Id].data.map(el => {
        const label = moment()
          .month(disbursement.month - 1)
          .year(disbursement.year)
          .format(DATE_FORMAT);

        if (moment(el.x).isSame(label, 'month')) {
          el.y = disbursement.totalQuantity;
        }

        el.tooltip = `${disbursement.productName}: ${el.y}`;
        el.x = moment(el.x).toDate();
        return el;
      });

      return acc;
    },
    { allIds: [], byId: {} }
  );

  normalized.colors = generateColors(normalized.allIds.length)

  normalized.allIds.forEach((id, i) => {
    normalized.byId[id].color = normalized.colors[i];
  })

  return normalized;
};
export  const handleFilterData = (id, filtered, disbursements) => {
  const index = filtered.ids.indexOf(id);
  const colors = [];
  let ids = [];
  if (index !== -1) {
    ids = filtered.ids.filter(el => {
      if (el !== id) {
        colors.push(disbursements.data.byId[el].color);
        return true;
      }
      return false;
    });
  } else {
    disbursements.data.allIds.forEach(el => {
      if (el === id || filtered.ids.includes(el)) {
        ids.push(el);
        colors.push(disbursements.data.byId[el].color);
      }
    });
  }

  return { colors, ids }
};