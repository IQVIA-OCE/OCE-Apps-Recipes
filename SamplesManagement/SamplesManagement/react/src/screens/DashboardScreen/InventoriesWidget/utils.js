import moment from 'moment';
import { INVENTORY_STATUS } from '../../InventoryScreen/constants';
import { INVENTORY_FORM_TYPE } from '../../../constants/Inventories';

export const normalizeInventories = recordTypesData => inventoriesData => {
  let data = {
    all: [],
  };
  if (
    recordTypesData &&
    recordTypesData.length &&
    inventoriesData &&
    inventoriesData.length
  ) {
    data = inventoriesData.reduce((acc, item) => {
      const type = recordTypesData.find(type => item.RecordTypeId === type.Id);
      if (!acc[type.DeveloperName]) {
        acc[type.DeveloperName] = [];
      }
      item.DeveloperName = type.DeveloperName;
      acc[type.DeveloperName].push(item);
      return acc;
    }, {});
    data.all = inventoriesData;
  }

  return data;
};

export const normalizeRecordTypes = recordTypesData => {
  let data = {
    all: [],
  };
  if (recordTypesData && recordTypesData.length) {
    data = recordTypesData.reduce((acc, item) => {
      acc[item.DeveloperName] = item;
      return acc;
    }, {});
    data.all = recordTypesData;
  }

  return data;
};

export const getMenuItems = (navigation, { inventories, recordTypes }) => {
  if (
    inventories &&
    inventories.InitialInventory &&
    inventories.InitialInventory.length
  ) {
    if (
      inventories.InitialInventory.some(
        el => el.OCE__Status__c === INVENTORY_STATUS.inProgress
      )
    ) {
      return [
        {
          text: 'Initial Inventory',
          onPress: () =>
            navigation.navigate('Inventory', {
              recordType: recordTypes.InitialInventory,
            }),
        },
      ];
    } else if (
      inventories.InitialInventory.some(
        el => el.OCE__Status__c === INVENTORY_STATUS.saved
      )
    ) {
      const item = inventories.InitialInventory.find(
        el => el.OCE__Status__c === INVENTORY_STATUS.saved
      );
      return [
        {
          text: 'Initial Inventory',
          onPress: () =>
            navigation.navigate('Inventory', {
              id: item.Id,
              type: INVENTORY_FORM_TYPE.editSaved,
            }),
        },
      ];
    } else {
      return [
        {
          text: 'Ad Hoc Inventory',
          onPress: () =>
            navigation.navigate('Inventory', {
              recordType: recordTypes.AdHocInventory,
            }),
        },
        {
          text: 'Audited Inventory',
          onPress: () =>
            navigation.navigate('Inventory', {
              recordType: recordTypes.AuditedInventory,
            }),
        },
        {
          text: 'Periodic Inventory',
          onPress: () =>
            navigation.navigate('Inventory', {
              recordType: recordTypes.PeriodicInventory,
            }),
        },
      ];
    }
  }
  return [
    {
      text: 'Initial Inventory',
      onPress: () =>
        navigation.navigate('Inventory', {
          recordType: recordTypes.InitialInventory,
        }),
    },
  ];
};

const DATE_FORMAT = 'MMM D, YYYY hh:mm A';

export const getDate = d => {
  const date = moment.utc(d);
  if (date.isValid()) {
    return date.format(DATE_FORMAT);
  } else {
    return moment.utc(d, 'DD-MM-YYYY hh:mm').format(DATE_FORMAT);
  }
};
