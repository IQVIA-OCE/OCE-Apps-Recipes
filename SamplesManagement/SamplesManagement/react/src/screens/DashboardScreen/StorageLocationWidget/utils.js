export const normalizeLocation = records =>
  records.length ? records[0]['OCE__FullAddress__c'] : '';
