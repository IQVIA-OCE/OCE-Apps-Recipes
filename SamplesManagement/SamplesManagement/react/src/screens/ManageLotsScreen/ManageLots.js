import React, { useEffect, useState } from 'react';
import { FlatList, View, StyleSheet, Text } from 'react-native';
import FormHeader from '../../components/FormHeader/FormHeader';
import LotItem from '../../components/LotItem/LotItem';
import { Banner } from 'apollo-react-native';
import { useBanner, useBoolean } from '../../hooks';
import { changeLotStatus, fetchLotsOffset } from '../../api/ManageLots';
import { normalizeLots } from '../DashboardScreen/ManageLotsWidget/utils';
import Loader from '../../components/Loader/Loader';

const keys = {
  CreatedDate: 'createdDate',
  Id: 'id',
  Name: 'name',
  LastModifiedDate: 'lastModified',
  OCE__Lot__r: 'lot',
  OCE__Lot__c: 'lotId',
  OCE__LotExpirationDate__c: 'expirationDate',
  OCE__Product__c: 'productId',
  OCE__Product__r: 'product',
  OCE__IsActive__c: 'isActive',
};
const ITEMS_ON_PAGE = 13;
const ManageLots = ({ navigation }) => {
  const [refreshWidget, refreshWidgetActions] = useBoolean(false);
  const [showActive, showActiveActions] = useBoolean(true);
  const [banner, setBanner] = useBanner();

  const [{ loading, error, data, page }, setValue] = useState({
    loading: true,
    error: '',
    data: null,
    page: 1,
    metadata: null,
  });

  const handleFetch = async () => {
    setValue(prevState => ({
      ...prevState,
      loading: true,
    }));

    try {
      let [data, metadata] = await fetchLotsOffset(ITEMS_ON_PAGE, showActive);

      setValue(prevState => ({
        ...prevState,
        loading: false,
        error: '',
        page: 1,
        data: normalizeLots(keys)(data),
        metadata,
      }));
    } catch (error) {
      setValue(prevState => ({
        ...prevState,
        loading: false,
        error: error,
        data: null,
        metadata: null,
      }));
    }
  };
  const handleLoadMore = async () => {
    try {
      let [newData, metadata] = await fetchLotsOffset(
        ITEMS_ON_PAGE,
        showActive,
        ITEMS_ON_PAGE * page
      );

      setValue(prevState => ({
        ...prevState,
        loading: false,
        error: '',
        data: [...data, ...normalizeLots(keys)(newData)],
        metadata,
        page: page + 1,
      }));
    } catch (error) {
      setValue(prevState => ({
        ...prevState,
        loading: false,
        error: error,
        data: null,
        metadata: null,
      }));
    }
  };
  const handleStatusChange = async (status, id) => {
    try {
      await changeLotStatus(status, id);
      refreshWidgetActions.setTrue();
      setBanner({
        variant: 'success',
        message: 'Success! Lot Allocation status has been updated.',
        visible: true,
        icon: 'checkbox-marked-circle',
      });
      await handleFetch();
    } catch (e) {
      setBanner({
        variant: 'error',
        message: e.message,
        visible: true,
        icon: 'alert-circle',
      });
    }
  };

  useEffect(() => {
    handleFetch();
  }, [showActive]);

  if (error) {
    return (
      <View>
        <Text>{error.message}</Text>
      </View>
    );
  }

  return (
    <View style={styles.root}>
      {loading && <Loader />}
      <Banner
        variant={banner.variant}
        icon={banner.icon}
        visible={banner.visible}
      >
        {banner.message}
      </Banner>
      <FormHeader
        title="Manage Lots"
        controls={[
          {
            label: 'Back',
            onPress: () =>
              navigation.navigate('Dashboard', {
                refreshLotsWidget: refreshWidget,
              }),
          },
          {
            label: `${showActive ? 'Show' : 'Hide'} Inactive Lots`,
            color: 'primary',
            mode: 'contained',
            onPress: () => {
              showActiveActions.toggle();
            },
          },
        ]}
      />

      <FlatList
        style={styles.list}
        data={data}
        refreshing={loading}
        onRefresh={handleFetch}
        ItemSeparatorComponent={() => <View style={styles.separator} />}
        renderItem={({ item, i }) => (
          <LotItem
            item={item}
            style={styles.item}
            handleStatusChange={status => handleStatusChange(status, item.id)}
          />
        )}
        onEndReached={handleLoadMore}
        onEndReachedThreshold={0.5}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
  list: {
    paddingHorizontal: 10,
  },
  separator: {
    borderBottomColor: '#dddbda',
    borderBottomWidth: 1,
  },
  item: {
    marginBottom: 0,
    paddingTop: 10,
    paddingBottom: 5,
  },
});

export default ManageLots;
