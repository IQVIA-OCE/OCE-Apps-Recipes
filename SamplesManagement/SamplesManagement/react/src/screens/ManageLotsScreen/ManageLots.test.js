import React from 'react';
import ManageLots from './ManageLots';
import { useBanner, useBoolean } from '../../hooks';
import * as api from '../../api/ManageLots';
import { normalizeLots } from '../DashboardScreen/ManageLotsWidget/utils';
import { FlatList } from 'react-native';
import { Button } from 'apollo-react-native';
import { render, act, waitFor } from '@testing-library/react-native';

jest.mock('../../hooks');

jest.mock('../DashboardScreen/ManageLotsWidget/utils');

const navigation = {
  navigate: jest.fn(),
};

const setBanner = jest.fn();
const useBannerValue = [
  { variant: '', message: '', visible: false, icon: '' },
  setBanner,
];
const data = [
  {
    name: 'LA-000778',
    lotId: 'a1Yf4000000kxHbEAI',
    lastModified: '2020-07-21T11:51:38.000+0000',
    lot: {
      name: 'L-0001',
      productId: 'a22f4000000kaTyAAI',
      product: {
        name: 'OKSF-2834_1',
      },
    },
    expirationDate: null,
    createdDate: '2019-06-22T22:05:40.000+0000',
    isActive: true,
    id: 'a2Af400000ZgLGKEA3',
  },
  {
    name: 'LA-000778',
    lotId: 'a1Yf4000000kxHbEAI',
    lastModified: '2020-07-21T11:51:38.000+0000',
    lot: {
      name: 'L-0003',
      productId: 'a22f4000000kaTyAAI',
      product: {
        name: 'OKSF-23434_2',
      },
    },
    expirationDate: null,
    createdDate: '2019-06-22T22:05:40.000+0000',
    isActive: true,
    id: 'a2Af400000ZgLGKEA2',
  },
];

let changeLotStatus, fetchLotsOffset;

describe('ManageLots', () => {
  beforeEach(() => {
    normalizeLots.mockImplementation(() => d => d);
    fetchLotsOffset = jest
      .spyOn(api, 'fetchLotsOffset')
      .mockReturnValueOnce([[data[0]]])
      .mockReturnValue([[data[1]]]);
    changeLotStatus = jest.spyOn(api, 'changeLotStatus').mockReturnValue();

    useBanner.mockReturnValue(useBannerValue);
    useBoolean.mockReturnValue([
      false,
      { toggle: jest.fn(), setFalse: jest.fn(), setTrue: jest.fn() },
    ]);
  });
  
  it('Should render without error', async () => {
    const { container } = render(<ManageLots navigation={navigation} />)
    
    act(() => container.findByType(FlatList).props.onEndReached());

    act(() =>
      container.findAllByType(Button)[0].props.onPress()
    );
    act(() =>
      container.findAllByType(Button)[1].props.onPress()
    );

    expect(navigation.navigate).toHaveBeenCalled();
    expect(fetchLotsOffset).toHaveBeenCalledTimes(2);
  });
});

describe('ManageLots error', () => {
  beforeEach(() => {
    normalizeLots.mockImplementation(() => d => d);
    jest
      .spyOn(api, 'fetchLotsOffset')
      .mockRejectedValue({ message: 'error message' });

    useBanner.mockReturnValue(useBannerValue);
    useBoolean.mockReturnValue([
      true,
      { toggle: jest.fn(), setFalse: jest.fn(), setTrue: jest.fn() },
    ]);
  });
  it('Should render with error', async () => {
    const { getByText } = render(<ManageLots navigation={navigation} />)

    await waitFor(() => {
      expect(getByText('error message')).toBeTruthy()
    })
  });
});
