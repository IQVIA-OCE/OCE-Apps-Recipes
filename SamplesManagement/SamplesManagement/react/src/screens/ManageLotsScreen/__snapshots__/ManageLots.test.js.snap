// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageLots Should render without error 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 1,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.5,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 0.75,
        "zIndex": 99,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "overflow": "hidden",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
      <View
        onLayout={[Function]}
        style={
          Object {
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "translateY": -0,
              },
            ],
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 0,
              "marginLeft": 16,
              "marginRight": 56,
              "marginTop": 8,
            }
          }
        >
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "margin": 8,
                    },
                    Object {
                      "color": "#000000",
                    },
                  ],
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "flex-end",
              "margin": 4,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F6F6F6",
        "borderColor": "#DADADA",
        "borderWidth": 1,
        "flexDirection": "row",
        "padding": 12,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 3,
            "height": 35,
            "marginRight": 10,
            "width": 35,
          },
          Object {
            "backgroundColor": "#AEAEAE",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 32,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï€„
      </Text>
    </View>
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 17,
            "fontWeight": "bold",
          }
        }
      >
        Manage Lots
      </Text>
    </View>
    <View />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingTop": 4,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#d9d9d9",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "elevation": 0,
            "marginLeft": 7,
            "minWidth": 64,
            "opacity": 1,
            "width": "auto",
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessibilityTraits="button"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-only"
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "borderRadius": 4,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "textAlign": "center",
                      },
                      undefined,
                      false,
                      Object {
                        "color": "#0768FD",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "marginHorizontal": 16,
                        "marginVertical": 6,
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Back
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "borderRadius": 0,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#0768FD",
            "borderRadius": 4,
            "borderStyle": "solid",
            "elevation": 2,
            "marginLeft": 7,
            "minWidth": 64,
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
            "width": "auto",
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessibilityTraits="button"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onLayout={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          pointerEvents="box-only"
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Array [
                  Object {
                    "borderRadius": 4,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "textAlign": "center",
                      },
                      undefined,
                      false,
                      Object {
                        "color": "#ffffff",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "marginHorizontal": 16,
                        "marginVertical": 6,
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Hide Inactive Lots
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "borderRadius": 0,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "createdDate": "2019-06-22T22:05:40.000+0000",
          "expirationDate": null,
          "id": "a2Af400000ZgLGKEA3",
          "isActive": true,
          "lastModified": "2020-07-21T11:51:38.000+0000",
          "lot": Object {
            "name": "L-0001",
            "product": Object {
              "name": "OKSF-2834_1",
            },
            "productId": "a22f4000000kaTyAAI",
          },
          "lotId": "a1Yf4000000kxHbEAI",
          "name": "LA-000778",
        },
        Object {
          "createdDate": "2019-06-22T22:05:40.000+0000",
          "expirationDate": null,
          "id": "a2Af400000ZgLGKEA2",
          "isActive": true,
          "lastModified": "2020-07-21T11:51:38.000+0000",
          "lot": Object {
            "name": "L-0003",
            "product": Object {
              "name": "OKSF-23434_2",
            },
            "productId": "a22f4000000kaTyAAI",
          },
          "lotId": "a1Yf4000000kxHbEAI",
          "name": "LA-000778",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "paddingHorizontal": 10,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 10,
                "width": "100%",
              },
              Object {
                "marginBottom": 0,
                "paddingBottom": 5,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 3,
                  "height": 32,
                  "justifyContent": "center",
                  "marginRight": 10,
                  "paddingTop": 2,
                  "width": 32,
                },
                Object {
                  "backgroundColor": "#f2cf5b",
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 28,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ïƒ–
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#0070d2",
                      "fontSize": 13,
                    }
                  }
                >
                  OKSF-2834_1
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "marginTop": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "width": "50%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#080707",
                        "fontSize": 12,
                      },
                      Object {
                        "color": "#3e3e3c",
                      },
                    ]
                  }
                >
                  Expiration:
                   
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#080707",
                        "fontSize": 12,
                      },
                      Object {
                        "color": "#080707",
                        "fontSize": 12,
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "width": "50%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#080707",
                        "fontSize": 12,
                      },
                      Object {
                        "color": "#3e3e3c",
                      },
                    ]
                  }
                >
                  Lot #:
                </Text>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#080707",
                          "fontSize": 12,
                        },
                        Object {
                          "color": "#0070d2",
                        },
                      ]
                    }
                  >
                    L-0001
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Array [
                  null,
                  undefined,
                ]
              }
            >
              <RCTSwitch
                accessibilityRole="button"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onTintColor="#00c221"
                style={
                  Array [
                    Object {
                      "height": 31,
                      "width": 51,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scale": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "",
                        "borderRadius": 16,
                      },
                    ],
                  ]
                }
                thumbTintColor="#ffffff"
                tintColor=""
                value={true}
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#dddbda",
              "borderBottomWidth": 1,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 10,
                "width": "100%",
              },
              Object {
                "marginBottom": 0,
                "paddingBottom": 5,
                "paddingTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 3,
                  "height": 32,
                  "justifyContent": "center",
                  "marginRight": 10,
                  "paddingTop": 2,
                  "width": 32,
                },
                Object {
                  "backgroundColor": "#f2cf5b",
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 28,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ïƒ–
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#0070d2",
                      "fontSize": 13,
                    }
                  }
                >
                  OKSF-23434_2
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "marginTop": 5,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "width": "50%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#080707",
                        "fontSize": 12,
                      },
                      Object {
                        "color": "#3e3e3c",
                      },
                    ]
                  }
                >
                  Expiration:
                   
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#080707",
                        "fontSize": 12,
                      },
                      Object {
                        "color": "#080707",
                        "fontSize": 12,
                      },
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "width": "50%",
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#080707",
                        "fontSize": 12,
                      },
                      Object {
                        "color": "#3e3e3c",
                      },
                    ]
                  }
                >
                  Lot #:
                </Text>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#080707",
                          "fontSize": 12,
                        },
                        Object {
                          "color": "#0070d2",
                        },
                      ]
                    }
                  >
                    L-0003
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Array [
                  null,
                  undefined,
                ]
              }
            >
              <RCTSwitch
                accessibilityRole="button"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                onTintColor="#00c221"
                style={
                  Array [
                    Object {
                      "height": 31,
                      "width": 51,
                    },
                    Array [
                      Object {
                        "transform": Array [
                          Object {
                            "scale": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "",
                        "borderRadius": 16,
                      },
                    ],
                  ]
                }
                thumbTintColor="#ffffff"
                tintColor=""
                value={true}
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;

exports[`ManageLots error Should render with error 1`] = `
<View>
  <Text>
    Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.
  </Text>
</View>
`;
